### PR TITLE
feat: add ExposeWithPermission decorator

### PR DESCRIPTION
**Pull request by edgarmueller**

### CODE REVIEW

File: example-apps/my-nest-project/src/app.controller.ts [hunk 1]
```diff
@@ -1,26 +1,46 @@
...
   @AuthAction([
-    ArticleAction.Admin.if(body(b => b.published)),
-    ArticleAction.Write.if(IsAuthor)
+    ArticleAction.Admin.if(body((b) => b.published)),
+    ArticleAction.Write.if(IsAuthor),
   ])
   @UsePipes(new ValidationPipe())
-  getHello(@Body() updateArtcileDto: UpdateArticleDto): string {
-    return this.appService.getHello();
+  @UseInterceptors(ExposeIfHasPermissionForInterceptor)
```
Comment (ottovw) [inline example-apps/my-nest-project/src/app.controller.ts:42]: PermissionInterceptor

File: example-apps/my-nest-project/src/get-article.dto.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { IsBoolean, IsNumber, IsString } from 'class-validator';
+import { ExposeIfHasPermissionFor } from './expose-if-has-permission-for.decorator';
+
+export class GetArticleDto {
+  @ExposeIfHasPermissionFor('read-article')
+  @IsString()
+  name: string;
+
+  @ExposeIfHasPermissionFor('write-article')
```
Comment (ottovw) [inline example-apps/my-nest-project/src/get-article.dto.ts:9]: ExposeWithPermission

File: example-apps/my-nest-project/src/expose-if-has-permission-for.interceptor.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  Scope,
+} from '@nestjs/common';
+import { ClassTransformOptions } from 'class-transformer';
+import { Observable, from } from 'rxjs';
+import { mergeMap, map } from 'rxjs/operators';
+import { REQUEST } from '@nestjs/core';
+import { PermissionProvider } from '@kittgen/nestjs-authorization';
+
+@Injectable({ scope: Scope.REQUEST })
+export class ExposeIfHasPermissionForInterceptor extends ClassSerializerInterceptor {
```
Comment (ottovw) [inline example-apps/my-nest-project/src/expose-if-has-permission-for.interceptor.ts:17]: WTF MATE?! Naming! ;)

**Review by ottovw — CHANGES_REQUESTED** [review]

File: example-apps/my-nest-project/src/expose-if-has-permission-for.interceptor.ts (hunk 3 above)
Comment (edgarmueller) [reply example-apps/my-nest-project/src/expose-if-has-permission-for.interceptor.ts:17]: 🤮

**Review by ottovw — APPROVED** [review]

**Comment by ottovw** [comment]
@edgarmueller 👍

**Pull request merged**